
- name: Double extended error message test case
  hosts: all
  gather_facts: no

  tasks:
    - name: Invalid task name
      # This task has an invalid task name, which should trigger an error
      fail:
        msg: "Invalid task name"

    - name: Invalid module name
      # This task uses an invalid module name, which should trigger an error
      invalid_module_name:
        arg1: "Hello"
        arg2: 123

    - name: Missing module argument
      # This task is missing a required argument for the file module, which should trigger an error
      file:
        path: "/etc/test"

    - name: Missing playbook argument
      # This task is missing a required argument for the include_vars module, which should trigger an error
      include_vars:

    - name: Invalid variable name
      # This task uses an invalid variable name, which should trigger an error
      debug:
        var: "{{ invalid variable name }}"

    - name: Invalid inventory hostname
      # This task uses an invalid hostname, which should trigger an error
      ping:
        data: "test"
      delegate_to: invalid_host

    - name: Invalid data type for a module argument
      # This task uses an invalid data type for a module argument, which should trigger an error
      file:
        path: "/etc/test"
        owner: 123

    - name: Invalid syntax for filter plugin
      # This task uses invalid syntax for a filter plugin, which should trigger an error
      debug:
        msg: "{{ 'test' | invalid_plugin }}"

    - name: Invalid include statement
      # This task uses an invalid syntax for an include statement, which should trigger an error
      include: "invalid_filename.yml"

    - name: Invalid loop syntax
      # This task uses an invalid syntax for a loop, which should trigger an error
      debug:
        msg: "{{ item invalid_list }}"
      loop: ["test", "123", "456"]
